### PR TITLE
check_format: switch to Python 3 and buildozer.

### PR DESCRIPTION
**Pull request by htuch**
* Update check_format.py etc. to Python 3. Cleaned up a bunch of adhoc file stuff to better handle
  common UTF-8 concerns that occur in the new world.

* Switch envoy_build_fixer.py to Buildozer for other transformers, undoing the crufty parser. As
  well as simplifying, it can now do things like removed unused loads.

Signed-off-by: Harvey Tuch <htuch@google.com>

### CODE REVIEW

File: tools/check_format.py [hunk 1]
```diff
@@ -146,6 +147,29 @@
...
+
+
+# Obtain all the lines in a given file.
+def readLines(path):
+  with open(path, encoding='utf-8', mode='r') as f:
+    return f.readlines()
+
+
+# Read a UTF-8 encoded file as a str.
+def readFile(path):
```
Comment (lizan) [inline tools/check_format.py]: consider use pathlib (which is standard library in python3) and https://docs.python.org/3/library/pathlib.html#pathlib.Path.read_text?

Comment (htuch) [reply tools/check_format.py]: Good idea, done.

**Comment by htuch** [comment]
@lizan @jmarantz is there any other feedback beyond the pathlib fix that I just made? Would be good to merge this to unblock the other PRs behind it when you folks are happy.

File: tools/check_format.py [hunk 2]
```diff
@@ -1,13 +1,14 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 
 from __future__ import print_function
```
Comment (lizan) [inline tools/check_format.py]: delete this line

File: tools/check_format.py [hunk 3]
```diff
@@ -153,6 +155,25 @@
 # yapf: enable
 
 
+# Map a line transformation function across each line of a file.
+# .bak temporaries.
+def replaceLines(path, line_xform):
+  # We used to use fileinput in the older Python 2.7 script, but this doesn't do
+  # inplace mode and UTF-8 in Python 3, so doing it the manual way.
+  output_lines = [line_xform(line) for line in readLines(path)]
+  pathlib.Path(path).write_text('\n'.join(output_lines), encoding='utf-8')
```
Comment (jmarantz) [inline tools/check_format.py]: i'll note that this needs to buffer the entire output set before writing, which seems wasteful of memory. But I guess this doesn't matter for this application?

Comment (lizan) [reply tools/check_format.py]: Yeah I don't think it matters, not likely to have MB level size of a source code file.

Comment (htuch) [reply tools/check_format.py]: Yep, that was my thinking as well.

**Pull request merged**